lib/Doctrine/ORM/Mapping/FieldMapping.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ArrayAccess;
+
+use function in_array;
+use function property_exists;
+
+/** @template-implements ArrayAccess<string, mixed> */
+final class FieldMapping implements ArrayAccess
+{
+    /** @var int|null The database length of the column. Optional. Default value taken from the type. */
+    public int|null $length = null;
+    /**
+     * @var bool|null Marks the field as the primary key of the entity. Multiple
```

Comment:
While trying to fix the build, I reverted to `bool|null` here, to be closer to the previous situation. Now that the build is green, I can do a separate commit to make boolean field not-nullable, and default to `false`.